src/ipfs.ts

```diff
@@ -330,10 +340,18 @@ class NoOpCache implements ObjectCache {
   putBinary(_cacheKey: string, _data: Uint8Array, _mimeType?: string | undefined): Promise<void> {
     return Promise.resolve()
   }
+  clearCache(_cacheKey: string): void {}
 }
 
 export class PinataStorage extends PinataStorageWithCache {
-  constructor(pinataToken: string) {
-    super(pinataToken, new NoOpCache())
+  constructor(pinataToken: string, ipfsGatewayOptions?: IPFSGatewayOptions) {
+    super(pinataToken, new NoOpCache(), ipfsGatewayOptions)
   }
 }
+
+export function getCIDUrl(ipfsGatewayBaseUrl: URL, cid: string) {
+  const copy = new URL(ipfsGatewayBaseUrl.toString())
+  copy.pathname = `/ipfs/${cid}`
```

Comment:
Not all gateways use this format, for example cloudflare used a subdomain.

I think it's probably ok to assume it's path based (not what cloudflare does), however I'd be inclined to require the full url minus the cid, so in the default value that'd be `https://ipfs.algonode.dev/ipfs`

Reply:
Cloudflare actually still worked with this format